### PR TITLE
roberto in da haus

### PR DESCRIPTION
**Pull request by kquirapas**

### CODE REVIEW

**Comment by Irizwaririz** [comment]
https://github.com/kquirapas/odin-project/blob/d19a4115e5bcea56eb4d86f39108ad812988c194/tictactoe/scripts/main.js#L10
https://github.com/kquirapas/odin-project/blob/d19a4115e5bcea56eb4d86f39108ad812988c194/tictactoe/scripts/main.js#L115
https://github.com/kquirapas/odin-project/blob/d19a4115e5bcea56eb4d86f39108ad812988c194/tictactoe/scripts/main.js#L13
https://github.com/kquirapas/odin-project/blob/d19a4115e5bcea56eb4d86f39108ad812988c194/tictactoe/scripts/main.js#L142

Since we're only pushing to `takenID` and `players` and not re-assigning them, it may be better to declare them as a `const` instead:

```javascript
const takenID = [];
```
```javascript
let players = [];
```

**Comment by Irizwaririz** [comment]
https://github.com/kquirapas/odin-project/blob/d19a4115e5bcea56eb4d86f39108ad812988c194/tictactoe/scripts/main.js#L3-L7
https://github.com/kquirapas/odin-project/blob/d19a4115e5bcea56eb4d86f39108ad812988c194/tictactoe/scripts/main.js#L12
https://github.com/kquirapas/odin-project/blob/d19a4115e5bcea56eb4d86f39108ad812988c194/tictactoe/scripts/main.js#L14
https://github.com/kquirapas/odin-project/blob/d19a4115e5bcea56eb4d86f39108ad812988c194/tictactoe/scripts/main.js#L15-L17
https://github.com/kquirapas/odin-project/blob/d19a4115e5bcea56eb4d86f39108ad812988c194/tictactoe/scripts/main.js#L20-L21
https://github.com/kquirapas/odin-project/blob/d19a4115e5bcea56eb4d86f39108ad812988c194/tictactoe/scripts/main.js#L94-L98

The variables above weren't re-assigned so it might be better to declare them as `const` instead

```javascript
const playerFactory = (name, piece) => {
```

```javascript
const boardSize = 3; 
```

```javascript
const view = document.getElementById("display");
const getBoardValue = (index) => {
    return document.getElementById(index).innerHTML;
};
```

```javascript
const p = players[currPlayer];
const pSymbol = p.getPiece();
```

```javascript
const showResult = (result) => {
    const overlay = document.getElementById("overlay");
    overlay.innerHTML = result;
    overlay.style.visibility = "visible";
};
```

**Comment by Irizwaririz** [comment]
https://github.com/kquirapas/odin-project/blob/d19a4115e5bcea56eb4d86f39108ad812988c194/tictactoe/scripts/main.js#L25
https://github.com/kquirapas/odin-project/blob/d19a4115e5bcea56eb4d86f39108ad812988c194/tictactoe/scripts/main.js#L42
https://github.com/kquirapas/odin-project/blob/d19a4115e5bcea56eb4d86f39108ad812988c194/tictactoe/scripts/main.js#L58
https://github.com/kquirapas/odin-project/blob/d19a4115e5bcea56eb4d86f39108ad812988c194/tictactoe/scripts/main.js#L70
https://github.com/kquirapas/odin-project/blob/d19a4115e5bcea56eb4d86f39108ad812988c194/tictactoe/scripts/main.js#L86
https://github.com/kquirapas/odin-project/blob/d19a4115e5bcea56eb4d86f39108ad812988c194/tictactoe/scripts/main.js#L127

It might be better to use a strict equality (`===`) for these conditional expressions instead so that it won't misfire if the two operands have different types

source: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Strict_equality

**Comment by Irizwaririz** [comment]
It might be better to use JavaScript Classes instead for better code organization. Plus, you could instantiate multiple instances of the game at once

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes

But other than that, great job! 💯

**Review by devofalltrades — COMMENTED** [review]
HOTDOG

File: tictactoe/scripts/main.js [hunk 1]
```diff
@@ -1,3 +1,4 @@
+// enjoy ka jc hahahaah
 "use strict"
 
 let playerFactory = (name, piece) => {
```
Comment (devofalltrades) [inline tictactoe/scripts/main.js:4]: hotdog

**Review by devofalltrades — COMMENTED** [review]
```
let playerFactory = (name, piece) => {
    const getName = () => name;
    const getPiece = () => piece;
    return {getName, getPiece};
};
```

HIOTDOG

**Review by devofalltrades — COMMENTED** [review]
https://github.com/kquirapas/odin-project/pull/1/files#diff-4101d6ac93e1ad929e8b33e17f0b2ca8d069c9264109a296ec20b138d2051f60R11-R14

**Review by devofalltrades — COMMENTED** [review]
https://github.com/kquirapas/odin-project/blob/d19a4115e5bcea56eb4d86f39108ad812988c194/tictactoe/scripts/main.js#L1-L14
https://github.com/kquirapas/odin-project/blob/d19a4115e5bcea56eb4d86f39108ad812988c194/tictactoe/scripts/main.js#L15-L16
https://github.com/kquirapas/odin-project/pull/1/files#diff-4101d6ac93e1ad929e8b33e17f0b2ca8d069c9264109a296ec20b138d2051f60L1-L12

**Pull request merged**